packages/generator/src/generator.ts

```diff
@@ -99,7 +105,16 @@ export abstract class Generator<T extends GeneratorOptions = GeneratorOptions> e
         const templateValues = await this.getTemplateValues()
 
         this.fs.copy(this.sourcePath(filePath), this.destinationPath(pathSuffix), {
-          process: (input) => this.process(input, pathSuffix, templateValues),
+          process: (input) => {
+            let formattedData = this.process(input, pathSuffix, templateValues)
+
+            if (prettierExtensions.test(pathSuffix) && typeof formattedData === 'string') {
```

Comment:
would you mind moving this logic into `this.process`? That'll avoid splitting our file manipulation into two different spots

Reply:
Since the `process` function need to be synchronous so I have to get the `prettier options` in the `run` function, is that Ok, do you have any suggestions, thank you.

Reply:
I don't believe `process` is overridden anywhere, you should be able to make `process` a `private` method and make it `async`, the only reason I made `process` synchronous was because there was nothing async inside, so feel free to change that!

Reply:
Sorry for my bad explanation, `mem-fs-editor` requires the `process` function to be synchronous so I cannot change it to be async, even if I try to do that then it will not work so I left the logic to get the prettier config in the `write` function.

Reply:
ohhhh i see. that's fine then!